src/lib/components/tree/node-tree/terra-node-tree.component.ts

```diff
@@ -131,39 +131,39 @@ export class TerraNodeTreeComponent<D> implements OnDestroy, OnInit
         return isVisible || hasVisibleChild;
     }
 
-    private _checkVisibility(node:TerraNodeInterface<D>, searchValue:string):boolean
+    private _matchesSearchString(node:TerraNodeInterface<D>, searchValue:string):boolean
     {
-        let hasValidCaptionOrTag:boolean = false;
+        return this._matchesName(node, searchValue) || this._matchesTags(node, searchValue);
+    }
 
-        let tags:Array<string> = node.tags;
+    private _matchesTags(node:TerraNodeInterface<D>, searchValue:string):boolean
+    {
+        const tags:Array<string> = node.tags;
```

Comment:
```suggestion
        const tags:Array<string> = node.tags || [];
```
So kannst du dir das if drunter sparen